dot/sync/syncer_test.go

```diff
@@ -42,6 +44,17 @@ func TestMain(m *testing.M) {
 	os.Exit(code)
 }
 
+type syncAPIMock struct {
+	fnGetHighestBlock func() (int64, error)
+}
+
+func (_m syncAPIMock) start()                                                         {}
+func (_m syncAPIMock) stop()                                                          {}
+func (_m syncAPIMock) setBlockAnnounce(from peer.ID, header *types.Header) error      { return nil }
+func (_m syncAPIMock) setPeerHead(p peer.ID, hash common.Hash, number *big.Int) error { return nil }
+func (_m syncAPIMock) syncState() chainSyncState                                      { return 0 }
+func (_m syncAPIMock) getHighestBlock() (int64, error)                                { return _m.fnGetHighestBlock() }
```

Comment:
I would suggest to use the generate command `//go:generate mockgen -destination=./mock_chain_sync_test.go -package $GOPACKAGE . ChainSync` (execute `go generate -run mockgen ./...`) to generate the `ChainSync` interface mock ([link](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md)), this command will create a new in the `./dot/sync` folder file with the function `NewMockChainSync` and at the function `TestHighestBlock`:

```go
t.Run(ts.name, func(t *testing.T) {
   ctrl := gomock.NewController(t)
   chainSyncMock := NewMockChainSync(ctrl)
   chainSyncMock.EXPECT().getHighestBlock().Return(ts.in.highestBlock, ts.in.err)
   ...
```

Reply:
Done in 2c5b83c